cpp/src/arrow/filesystem/s3_test_util.cc

```diff
@@ -69,6 +76,47 @@ std::string MinioTestServer::access_key() const { return impl_->access_key_; }
 
 std::string MinioTestServer::secret_key() const { return impl_->secret_key_; }
 
+std::string MinioTestServer::ca_path() const {
+  return impl_->temp_dir_ca_->path().ToString();
+}
+
+Status MinioTestServer::GenerateCertificateFile() {
+  // create the dedicated folder for certificate file, rather than reuse the data
+  // folder, since there is test case to check whether the folder is empty.
+  ARROW_ASSIGN_OR_RAISE(impl_->temp_dir_ca_, TemporaryDir::Make("s3fs-test-ca-"));
+
+  ARROW_ASSIGN_OR_RAISE(auto public_crt_file,
+                        PlatformFilename::FromString(ca_path() + "/public.crt"));
+  ARROW_ASSIGN_OR_RAISE(auto public_cert_fd, FileOpenWritable(public_crt_file));
+  ARROW_RETURN_NOT_OK(FileWrite(public_cert_fd.fd(),
+                                reinterpret_cast<const uint8_t*>(kMinioCert),
+                                strlen(kMinioCert)));
+  ARROW_RETURN_NOT_OK(public_cert_fd.Close());
+
+  ARROW_ASSIGN_OR_RAISE(auto private_key_file,
+                        PlatformFilename::FromString(ca_path() + "/private.key"));
+  ARROW_ASSIGN_OR_RAISE(auto private_key_fd, FileOpenWritable(private_key_file));
+  ARROW_RETURN_NOT_OK(FileWrite(private_key_fd.fd(),
+                                reinterpret_cast<const uint8_t*>(kMinioPrivateKey),
+                                strlen(kMinioPrivateKey)));
+  ARROW_RETURN_NOT_OK(private_key_fd.Close());
+
+  // Set the trusted CA certificate
+#if defined(__linux__)
+  arrow::fs::FileSystemGlobalOptions global_options;
+  global_options.tls_ca_dir_path = ca_path();
+  ARROW_RETURN_NOT_OK(arrow::fs::Initialize(global_options));
+#elif defined(_WIN32)
+  // Windows does not have a standard location for CA certificates
+  auto import_cert_process = std::make_unique<util::Process>();
+  ARROW_RETURN_NOT_OK(import_cert_process->SetExecutable("certutil"));
+  import_cert_process->SetArgs(
+      {"-addstore", "-f", "ArrowTest", public_crt_file.ToString()});
+  ARROW_RETURN_NOT_OK(import_cert_process->Execute());
```

Comment:
Hmm. This changes the global configuration, right?

This is not used because we use `verifySSL = false`, right?

Reply:
@kou the verifySSL = false is ONLY set in the test code, and used for a part of the test cases like TestMinioServer-> Connect,etc.
however, for the other test cases that invoked the workflow in arrow(the client is created via https://github.com/apache/arrow/blob/main/cpp/src/arrow/filesystem/s3fs.cc#L1098-L119), the  verifySSL = false could NOT be set.  
Since the capath solution only work for Linux, do you think it's acceptable to Run the https case in Linux and HTTP case for the other platform,Thanks
cc @pitrou

Reply:
@pitrou Any insights/suggestions on this? Thanks

Reply:
OK. Let's use the workaround (use HTTPS only on Linux) in this PR.

It seems that we can solve it by adding `verify_tls` to `arrow::fs::FileSystemGlobalOptions` or `arrow::fs::S3Options`.

Reply:
Would `tls_ca_file_path` work on Windows?

Reply:
@pitrou I think tls_ca_file_path would also NOT work on windows,  according to https://docs.aws.amazon.com/sdk-for-cpp/v1/developer-guide/client-config.html caPath, caFile only work on linux. 
but if we could expose the verify_tls setting on arrow::fs::FileSystemGlobalOptions or arrow::fs::S3Options as @kou  suggested, then we could also run the test case on windows.

Reply:
@kou  Even for the linux, the capath solution would require the folder prepared with the OpenSSL `c_rehash `utility, so it would require the c_rehash tool installed in the test server. In order to reduce the complex, i tend to set the verfiySSL to false for simple, can you help to review again?Thanks